astropy/io/fits/connect.py

```diff
@@ -112,7 +112,8 @@ def _decode_mixins(tbl):
 
 
 def read_table_fits(input, hdu=None, astropy_native=False, memmap=False,
-                    character_as_bytes=True, unit_parse_strict='warn'):
+                    character_as_bytes=True, unit_parse_strict='warn',
+                    mask_invalid=True):
```

Comment:
Is it worth considering making `mask_invalid` default to  `False` if `memmap is True`?

Reply:
I'm not sure, it can be unexpected to get an unmasked table when using `memmap=True`. Maybe a warning if `memmap=True` and `mask_invalid=True` to explain the user why they also need to set `mask_invalid=False` ?

Reply:
Both seem to have advantages: I like the clarity of the warning in that users do not get unexpected data, but worry slightly about `read_table_fits` being done well inside code where a user may not have access.  

I guess eventually we could have a `memmap='if_possible'`...

I do worry a bit how this function is getting more and more arguments about arguably small differences in how to handle specific items. I wonder if this is the best way to handle them. But that's for another discussion...

Reply:
The interest of using memmap here is limited to quite specific cases (reading a subset of the rows,  #9404, https://github.com/astropy/astropy/issues/8573#issuecomment-482561218), so I don't think a warning would annoy a lot of people. But it's also true if we set `mask_invalid=False` when using memmap, so maybe it's a good option as well.

Reply:
I'll let you make the call -- I cannot decide even which option I think is better...

Reply:
Ok, I went for the first option, setting `mask_invalid=False` when using memmap, and added some explanations in the docstring.